test/e2e/custominjection_test.go

```diff
@@ -0,0 +1,531 @@
+package e2e
+
+import (
+	"time"
+
+	"github.com/k8snetworkplumbingwg/network-resources-injector/test/util"
+
+	cniv1 "github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/apis/k8s.cni.cncf.io/v1"
+	corev1 "k8s.io/api/core/v1"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("Verify 'User Defined Injections'", func() {
+	var pod, pod2 *corev1.Pod
+	var configMap *corev1.ConfigMap
+	var nad, nad2 *cniv1.NetworkAttachmentDefinition
+	var err error
+
+	Context("Positive use cases - expected that NRI will inject correctly custom definitions", func() {
+		BeforeEach(func() {
+			nad = util.GetResourceSelectorOnly(testNetworkName, *testNs, testNetworkResName)
+			err = util.ApplyNetworkAttachmentDefinition(networkClient.K8sCniCncfIoV1Interface, nad, timeout)
+			Expect(err).Should(BeNil())
+
+			// second network attachment that is used by user custom injections
+			nad2 = util.GetResourceSelectorOnly("sriov-net-attach-def", *testNs, "example.com/boo")
+			err = util.ApplyNetworkAttachmentDefinition(networkClient.K8sCniCncfIoV1Interface, nad2, timeout)
+			Expect(err).Should(BeNil())
+		})
+
+		AfterEach(func() {
+			util.DeletePod(cs.CoreV1Interface, pod, timeout)
+			util.DeleteNetworkAttachmentDefinition(networkClient.K8sCniCncfIoV1Interface, testNetworkName, nad, timeout)
+			util.DeleteNetworkAttachmentDefinition(networkClient.K8sCniCncfIoV1Interface, "sriov-net-attach-def", nad2, timeout)
+
+			if nil != configMap {
+				util.DeleteConfigMap(cs.CoreV1Interface, configMap, timeout)
+				configMap = nil
+			}
+		})
+
+		It("Config map in correct namespace, one label to inject, POD specification WITHOUT resource name", func() {
+			configMap = util.GetConfigMap("nri-user-defined-injections", "kube-system")
```

Comment:
I guess we shall wait for the controlSwitch configmap PR to merge first, then update this configMpa name and the following test cases? Or do you prefer the reverse order?

Reply:
Yes, I think that we can wait with this change for the second PR merge. For sure e2e code may require some adaptations.

Reply:
Solved. I did all adaptations.